src/commands/utils/uploadUDF.ts

```diff
@@ -125,5 +166,60 @@ export async function promptForUDFUploadParams(): Promise<UDFUploadParams | unde
     return undefined;
   }
 
-  return { environment: environment.id, cloud, region, artifactName, fileFormat };
+  return {
+    environment: environment.id,
+    cloud,
+    region,
+    artifactName,
+    fileFormat,
+    selectedFile: selectedFile ? selectedFile : undefined,
+  };
+}
+
+export async function handleUploadFile(
+  params: UDFUploadParams,
+  presignedURL: string,
+): Promise<void> {
+  if (!params.selectedFile) {
+    showErrorNotificationWithButtons("No file selected for upload.");
+    return;
+  }
+  const { file, blob, contentType } = await prepareUploadFileFromUri(params.selectedFile);
+
+  logger.info(`Starting file upload for cloud provider: ${params.cloud}`, {
+    artifactName: params.artifactName,
+    environment: params.environment,
+    cloud: params.cloud,
+    region: params.region,
+    contentType,
+  });
+
+  switch (params.cloud) {
+    // TODO: TS ENUMS FOR CLOUD PROVIDERS
+    case "Azure": {
+      logger.debug("Uploading to Azure storage");
+      const response = await uploadFileToAzure({
```

Comment:
To communicate start and end of streams we have these logger.debug message. One thought is that it takes a while and we've not even made the `createArtifact` call yet. Are these enought to communicate loading state to user...